app/controllers/embedded_terraform_credential_controller.rb

```diff
@@ -0,0 +1,148 @@
+class EmbeddedTerraformCredentialController < ApplicationController
+  before_action :check_privileges
+  before_action :get_session_data
+
+  after_action :cleanup_action
+  after_action :set_session_data
+
+  include Mixins::GenericListMixin
+  include Mixins::GenericSessionMixin
+  include Mixins::GenericShowMixin
+  include Mixins::BreadcrumbsMixin
+
+  menu_section :embedded_terraform_credentials
+
+  def self.display_methods
+    %w[repositories]
+  end
+
+  def self.model
+    ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential
+  end
+
+  def display_repositories
+    nested_list(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptSource)
+  end
+
+  def show_searchbar?
+    true
+  end
+
+  def button
+    case params[:pressed]
+    when "embedded_terraform_repositories_reload"
+      javascript_redirect(:action => 'show', :id => params[:id], :display => 'repositories')
+    when 'embedded_configuration_script_source_refresh' # refresh repositories from nested list
+      repository_refresh
+    when 'embedded_automation_manager_credentials_add'
+      javascript_redirect(:action => 'new')
+    when 'embedded_configuration_script_source_add' # add repository from nested list
+      javascript_redirect(:controller => 'embedded_terraform_repository', :action => 'new')
+    when 'embedded_automation_manager_credentials_edit'
+      javascript_redirect(:action => 'edit', :id => params[:miq_grid_checks])
+    when 'embedded_configuration_script_source_edit' # edit repository from nested list
+      javascript_redirect(:controller => 'embedded_terraform_repository', :action => 'edit', :id => params[:miq_grid_checks])
+    when 'ansible_credential_tag'
+      tag(self.class.model)
+    when "ansible_repository_tag" # repositories from nested list
+      tag(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptSource)
```

Comment:
Do each of these cases automatically do assert_privileges checks from `tag`, `javascript_redirect`, etc.?

Reply:
🤣 I think they do... looking down below.